scripts/ingests/utils.py

```diff
@@ -721,26 +741,57 @@ def ingest_parallaxes(db, sources, plxs, plx_errs, plx_refs, save_db=False, verb
 
     n_added = 0
 
-    for i, source in enumerate(sources):
+    for i, source in enumerate(sources): # loop through sources with parallax data to ingest
         db_name = db.search_object(source, output_table='Sources')[0]['source']
 
-        # Search for existing parallax data.
+        # Search for existing parallax data and determine if this is the best
         # If no previous measurement exists, set the new one to the Adopted measurement
-        # TODO: figure out adopted in cases of multiple measurements.
         adopted = None
+        duplicate = False
         source_plx_data = db.query(db.Parallaxes).filter(db.Parallaxes.c.source == db_name).table()
+        # TODO: Write a function to deal with setting adopted flag
         if source_plx_data is None or len(source_plx_data) == 0:
+            # if there's no other measurements in the database, set new data Adopted = True
             adopted = True
+            duplicate = False
+            old_adopted = None
+        elif len(source_plx_data) > 0:  # plx data already exists
+            for plx_data in source_plx_data:
+                if plx_data['reference'] == plx_refs[i]:  # check to see if other measurement is a duplicate of the new data
+                    duplicate = True
+                    verboseprint("Duplicate measurement\n", plx_data)
+                if plx_data['adopted'] == 1:  # check if another measurement is adopted
+                    old_adopted = plx_data
```

Comment:
You should not need to loop over each result to match references/adopted. Something like `ind = plx_data['reference'] == plx_refs[i]` should give you an array of True/False that you can sum up to see if any matches were found. Similarly for the checking of adopted check, which you can then use with `old_adopted = plx_data[ind]` to store that matched row.

Reply:
Ah, excellent -- thanks! Implemented.